projects/canopy/src/lib/forms/date/date-field.component.ts

```diff
@@ -144,30 +144,30 @@ export class LgDateFieldComponent implements OnInit, ControlValueAccessor, OnDes
       this.ngControl.control.updateValueAndValidity();
     }
 
-    this.valueChanges = this.dateFormGroup.valueChanges.subscribe((date: DateField) => {
-      const day = date.date ? ('0' + date.date).slice(-2) : '';
-      const month = date.month ? ('0' + date.month).slice(-2) : '';
-      const year = date.year ? date.year : '';
-      this.onChange(`${year}-${month}-${day}`);
-    });
+    this.subscriptions.push(
+      this.dateFormGroup.valueChanges.subscribe((date: DateField) => {
+        const day = date.date ? ('0' + date.date).slice(-2) : '';
+        const month = date.month ? ('0' + date.month).slice(-2) : '';
+        const year = date.year ? date.year : '';
+        this.onChange(`${year}-${month}-${day}`);
+      }),
 
-    // submit the group when the parent form is submitted
-    this.parentFormGroupDirective.ngSubmit
-      .pipe(filter(({ type }) => type === 'submit'))
-      .subscribe((event) => {
-        this.formGroupDirective.onSubmit(event);
-        this.ngControl.control.updateValueAndValidity();
-      });
+      // submit the group when the parent form is submitted
+      this.parentFormGroupDirective.ngSubmit
```

Comment:
Makes sense, so we weren't unsubscribing from this one? Should it have a test?

Reply:
I've not come across unit tests for unsubscribes before, but I suppose could by spying on the unsubscribe method after calling `ngOnDestroy` in the unit test? 🤔

Reply:
To be honest I have never tested unsubscribtions but i can try 🤷‍♀️